Sources/PubNub/Helpers/PubNubAPNSPayload.swift

```diff
@@ -0,0 +1,383 @@
+//
+//  PubNubAPNSPayload.swift
+//
+//  PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+//  Copyright © 2019 PubNub Inc.
+//  https://www.pubnub.com/
+//  https://www.pubnub.com/terms
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+// swiftlint:disable line_length
+
+import Foundation
+
+/// The PubNub push configuration payload
+public struct PubNubPushConfig: Codable, Hashable {
+  /// The targets of a published PubNub notification
+  public let targets: [PubNubPushTarget]
+  /// Version of the PubNub push configuration
+  public let version: String = "v2"
+  /// APS collapse id
+  public let collapseID: String?
+  /// Expiration time of the remote notification
+  public let expiration: Date?
+
+  enum CodingKeys: String, CodingKey {
+    case collapseID = "collapse_id"
+    case expiration
+    case targets
+    case version
+  }
+
+  public init(
+    targets: [PubNubPushTarget],
+    collapseID: String? = nil,
+    expiration: Date? = nil
+  ) {
+    self.targets = targets
+    self.collapseID = collapseID
+    self.expiration = expiration
+  }
+}
+
+/// The target of a published PubNub notification
+public struct PubNubPushTarget: Codable, Hashable {
+  /// The topic of the device
+  public let topic: String
+  /// The APS environment
+  public let environment: PushRouter.Environment
+  /// Devices that should not receive this notification
+  ///
+  /// This is likely the senders device token
+  public let excludedDevices: [String]?
+
+  enum CodingKeys: String, CodingKey {
+    case environment
+    case excludedDevices = "excluded_devices"
+    case topic
+  }
+
+  public init(
+    topic: String,
+    environment: PushRouter.Environment = .production,
```

Comment:
At least in Objective-C ticket default environment suggested to be set to “develop”

Reply:
From the customer perspective the deployed use-case would be setting it to production, and then develop would be used primarily only during development.

Reply:
I think this was a product requirement to set it to `development` by default.

Reply:
The default can be removed.  My point is that setting the default to development is equivalent to not having a default, because they'll always have to set it to production